docs/source/code.rst

```diff
@@ -0,0 +1,59 @@
+Code Guidelines
+-------------------------------------------
+
+Developers of eMach should read this document to understand expectations of code contributed to the eMach codebase. This document outlines the 
+code requirements for ``analyzers`` and ``machine_designs``. Understanding of proper Python coding practices are necessary to contribute to each 
+section of ``eMach``.
+
+Analyzers
+++++++++++++++++++++++++++++++++++++++++++++
+
+Analyzer modules should be located within ``mach_eval/analyzers`` and then placed within the appropriate subdirectory.
+
+An analyzer module may contain multiple analyzers if they are interdependent. Each analyzer must consist of the following classes:
+
+1. At least one Problem class  
+2. One Analyzer class
+3. Results class
+
+Problem Class
+*******************************************
+
+The purpose of the problem class is to provide the analyzer class the necessary information to analyze a problem. Aspects to consider:
+
+1. Naming
+    a. The problem class name should begin similarly to the analyzer class's name, i.e. ``ReallyGreatProblem`` for ``ReallyGreatAnalyzer``.
+2. Code comments on user input
+    a. Provide short description of each argument 
+    b. Specify argument units
+3. Recommended practices
+    a. Provide user data through the problem class initalizer 
+    b. Multiple problem classes can be defined for use with a single analyzer so that users can provide differing formats or conceptualizations of the necessary information (perhaps even requiring some computation within the problem class itself prior to being used in the analyzer).
+
+
+
+Analyzer Class
+*******************************************
+
+The Analyzer class is where the analysis operation is expected to occur. Aspects to consider:
+
+1. Required functions:
+    a. Each analyzer must provide an ``analyze(problem p)`` function that takes exactly one argument, a problem object, and returns the analysis results.
```

Comment:
This needs to be updated to provide a results object.

Reply:
I'm confused here, I thought the whole point of adding the results _class_ was to take the `analysis results` referenced in this snippet as inputs and return a single results _object_. I don't see how there should be a results object referenced in this section.

Reply:
That's correct. The `analyze()` function should now return a Results object which encapsulates all of the analysis results.